_posts/2016-05-11-android-testing-with-kotlin.md

```diff
@@ -0,0 +1,422 @@
+    }
+}
+
+dependencies {
+    ...
+    androidTestCompile 'com.android.support.test.espresso:espresso-core:2.2.2'
+}
+```
+
+Now to the test suite itself. Our example Espresso test class
+[`EspressoKotlinTest`](https://github.com/GreenhouseCI/kotlin-tests/blob/master/app/src/androidTest/kotlin/com/greenhouseci/kotlin_tests/kotlintests/EspressoKotlinTest.kt)
+looks like
+
+```ruby
+@RunWith(AndroidJUnit4::class)
+class EspressoKotlinTest {
+
+    ...
+
+    @Rule @JvmField
+    var activityRule = ActivityTestRule<MainActivity>(MainActivity::class.java)
+
+    ...
+
+    @Test
+    fun testAddItemsToList() {
+        val formatter = Formatter()
+        val lv = onData(anything()).inAdapterView(withId(R.id.lv_items))
+
+        var items = arrayOf(
+            Pair("    item 1", "item 1"),
+            Pair("item 2  ", "item 2"),
+            Pair("  item  \n3  ", "item 3")
+        )
+
+        items.map { pair -> addItemToList(pair.first) }
+        items.mapIndexed { index, pair ->
+            val expected = formatter.strip(pair.second)
+            lv.atPosition(index).onChildView(withId(R.id.tv_list_item)).check(
+                matches(withText(expected))
+            )
+        }
+    }
+}
+```
+
+You can see that the Espresso test suite at hand doesn't differ much from the
+unit tests we discussed in the previous section. And neither does it differ much from
+your already existing Espresso tests written in Java. Truth is, that majority of the
+code that does the heavy lifting is plain *Java* and other parts are almost
+drop in replacements.
+
+As usual with instrumentation tests, we start off by creating an instrumented JUnit 4
+test class by adding
+
+```ruby
+@RunWith(AndroidJUnit4::class)
+```
+
+annotation at the beginning of our test class definition. And naturally we can make
+use of Espresso's built-in `ActivityTestRule` to handle our activity life cycle:
+
+```ruby
+@Rule @JvmField
+var activityRule = ActivityTestRule<MainActivity>(MainActivity::class.java)
+```
+
+With one minor caveat that you need to annotate it with `@JvmField` to convert it
+from Kotlin a *property*, that is unrecognized by JUnit, to a JVM field that is edible
+by JUnit.
+
+Apart from these tiny details, you can just use the familiar Espresso API for the
+test functions.
+
+## Running our Kotlin tests
+
+Since we have already updated our `sourceSets` in `build.gradle` we don't have to
+go through any configuration magic any more. You can execute both unit and instrumentation
+tests right from Android Studio by just right-clicking either on your tests packages or
+Kotlin test files and clicking on "Run 'Tests in ...'" as can be seen below:
+
+<div style="text-align:center; padding: 20px 0;"><img src="/assets/run-kotlin-tests-form-android-studio.png" alt="Run Kotlin tests from Android Studio"/></div>
+
+The very same minimal configuration logic applies also in case you should want to
```

Comment:
"**also** applies ~~also in case you~~ should **you** want to execute your tests from **the** command line"